nixos/modules/services/display-managers/greetd.nix

```diff
@@ -102,12 +102,18 @@ in
 
     systemd.defaultUnit = "graphical.target";
 
+    # Create directories potentially required by supported greeters
+    # See https://github.com/NixOS/nixpkgs/issues/248323
+    systemd.tmpfiles.rules = [
+      "d '/var/cache/tuigreet' - greeter greeter - -"
```

Comment:
It might also be good to use `750` instead of `755` for the directory's permissions, idk